### PR TITLE
Rename the drain state to draining (Fixes #30833)

### LINKED ISSUE #30833: HaProxy drain mode 'bool' object is not callable error
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
HaProxy Module

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/Users/stewart/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ansible
  executable location = /Library/Frameworks/Python.framework/Versions/2.7/bin/ansible
  python version = 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 12:39:47) [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]

```

##### CONFIGURATION
`ansible-config dump --only-changed` returned nothing. I have changed no config.

##### OS / ENVIRONMENT
Running from OSX 10.10.5
Trying to control HaProxy 1.5.8 on Debian 8

##### SUMMARY
I am able to enable and disable HaProxy backends correctly but when I try to use the "drain" state type I get this error. The playbook works when I use enabled and disabled but fails with the below error when I use drain.

It says, `'bool' object is not callable`. I might just be configuring this wrong but I believe I am doing everything correctly and using the correct Ansible version.

##### STEPS TO REPRODUCE

```yaml
- name: Set server to drain
  hosts: haproxy
  sudo: yes
  tasks:
    - name: Set server1 to disabled
      haproxy:
        state: drain
        host: server1
        socket: /var/run/haproxy.sock
        backend: cluster_www
```

##### EXPECTED RESULTS

I expect the HaProxy backend to change to the drain state.

##### ACTUAL RESULTS

```
fatal: [10.5.59.15]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Shared connection to 10.5.59.15 closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_Xzs48s/ansible_module_haproxy.py\", line 454, in <module>\r\n    main()\r\n  File \"/tmp/ansible_Xzs48s/ansible_module_haproxy.py\", line 450, in main\r\n    ansible_haproxy.act()\r\n  File \"/tmp/ansible_Xzs48s/ansible_module_haproxy.py\", line 410, in act\r\n    self.drain(self.host, self.backend)\r\nTypeError: 'bool' object is not callable\r\n", "msg": "MODULE FAILURE", "rc": 0}
```

### PR DESCRIPTION
**Pull request by drmrd**
##### SUMMARY

This PR is meant to fix with the `haproxy` module reported in #30833. At present, the module is assigned a `drain` variable and a `drain` function. The solution proposed in this PR is to rename the function and associated value of the `state` module parameter to `draining`. This interface change is justified since it accomplishes both of the following goals:

1. It avoids the name collision producing bug #30833.
2. The present tense in `state: draining` makes more clear that declaring this state does not cause the module to wait until backend hosts finish draining before finishing its run.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

`haproxy` module

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/Users/drmrd/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/drmrd/Library/Python/2.7/lib/python/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

### CODE REVIEW

**Comment by ravibhure** [comment]
👍  shipit

**Review by TijnLambrechtsen — APPROVED** [review]
Works like a charm! Thnx a million ;-)

**Comment by bjolivot** [comment]
great work 
shipit

**Comment by narkq** [comment]
Works for me 👍
shipit please

**Comment by drmrd** [comment]
bot_status

**Comment by ravibhure** [comment]
shipit

**Comment by drmrd** [comment]
@akostyuk @alcamie101 @andreaso @briceburg @drcapulet @jpmens @jtyr @mgruener @nerzhul @ogenstad @resmo: I believe you are the other maintainers in this namespace apart from @ravibhure. Would one of you be willing to review this PR? TIA!

**Comment by skurylo** [comment]
Another bit is missing form this PR

On line 353
https://github.com/ansible/ansible/pull/30947/files#diff-68964ab5461607ca44a8ec589b71891bR353

If you're trying to DRAIN, the state will never be MAINT when `wait=yes`:
`if not self.drain or (state[0]['scur'] == '0' and state == 'MAINT'):`

So instead I did
`if not self.drain or (state[0]['scur'] == '0' and state[0]['status'] in ('MAINT', 'DRAIN')):`

**Comment by ravibhure** [comment]
Thanks @skurylo,
@drmrd can you please check and fix in your PR

**Comment by drmrd** [comment]
@skurylo Good catch! I'll push an update.

**Comment by drmrd** [comment]
@ravibhure I've pushed an update that should resolve the issue @skurylo found, and the PR is ready for review.

**Comment by ravibhure** [comment]
+1

**Comment by drmrd** [comment]
@ravibhure Do you know what else needs to be done (or who else needs to approve it) to get this PR merged?

**Comment by ravibhure** [comment]
Hi @sivel , when can we merge this fix.

**Comment by BobbyBakes** [comment]
@ravibhure @sivel I've really been looking forward to this fix. Is there any update regarding when this will be fixed?

**Comment by aramalipoor** [comment]
@drmrd any updates on this?

**Comment by ravibhure** [comment]
@sivel do you want this to be further change or any comments, I think this PR is fixed and ready for merge, please take your decision or discard if you don't feel this is relevant.
cc @drmrd

File: lib/ansible/modules/net_tools/haproxy.py [hunk 1]
```diff
@@ -198,7 +199,7 @@
 
 DEFAULT_SOCKET_LOCATION = "/var/run/haproxy.sock"
 RECV_SIZE = 1024
-ACTION_CHOICES = ['enabled', 'disabled', 'drain']
+ACTION_CHOICES = ['enabled', 'disabled', 'draining']
```
Comment (jorijinnall) [inline lib/ansible/modules/net_tools/haproxy.py:202]: I think it breaks backward compatibility for using this module

**Comment by jorijinnall** [comment]
I propose this pull request : https://github.com/ansible/ansible/pull/46478

**Comment by mscherer** [comment]
@jorijinnall is right, if the interface is changed, it would requires a changelog, ideally handling backward compatibility. I think that #46478 is better since smaller and backward compatible.

I also wonder why the bot didn't merge

**Comment by mscherer** [comment]
bot_status

**Comment by drmrd** [comment]
Thanks @jorijinnall for getting an alternative fix to this merged! I'm closing this PR now that it's no longer necessary.

**Pull request closed without merging**